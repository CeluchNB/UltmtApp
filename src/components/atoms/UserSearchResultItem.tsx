import * as React from 'react'
import { IconButton } from 'react-native-paper'
import { useColors } from '../../hooks'
import {
    ActivityIndicator,
    StyleSheet,
    Text,
    TouchableOpacity,
    View,
} from 'react-native'
import { size, weight } from '../../theme/fonts'

interface UserSearchResultItemProps {
    name: string
    username: string
    loading: boolean
    onPress: () => {}
}

const UserSearchResultItem: React.FC<UserSearchResultItemProps> = ({
    name,
    username,
    loading,
    onPress,
}) => {
    const { colors } = useColors()

    const styles = StyleSheet.create({
        container: {
            display: 'flex',
            flexDirection: 'row',
            margin: 5,
        },
        nameContainer: {
            flex: 1,
            alignSelf: 'center',
        },
        name: {
            color: colors.gray,
            fontSize: size.fontMedium,
            fontWeight: weight.full,
        },
        username: {
            color: colors.textSecondary,
            fontSize: size.fontFifteen,
            fontWeight: weight.full,
        },
        buttonStyle: {
            alignItems: 'center',
            justifyContent: 'center',
        },
    })

    return (
        <View>
            <TouchableOpacity
                style={styles.container}
                onPress={onPress}
                disabled={loading}>
                <View style={styles.nameContainer}>
                    <Text style={styles.name}>{name}</Text>
                    <Text style={styles.username}>{username}</Text>
                </View>
                {!loading && (
                    <IconButton
                        style={styles.buttonStyle}
                        color={colors.success}
                        icon="plus"
                        onPress={onPress}
                    />
                )}
                {loading && (
                    <ActivityIndicator
                        size="small"
                        color={colors.gray}
                        animating={loading}
                    />
                )}
            </TouchableOpacity>
        </View>
    )
}

export default UserSearchResultItem
