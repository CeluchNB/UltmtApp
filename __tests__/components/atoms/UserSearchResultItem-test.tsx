import '@testing-library/jest-native/extend-expect'
import * as React from 'react'
import renderer from 'react-test-renderer'
import UserSearchResultItem, {
    UserSearchResultItemProps,
} from './../../../src/components/atoms/UserSearchResultItem'
import { fireEvent, render } from '@testing-library/react-native'

let props: UserSearchResultItemProps

beforeEach(() => {
    props = {
        name: 'first last',
        username: 'firstlast',
        loading: false,
        onPress: jest.fn(),
        error: undefined,
    }
})

it('test matches snapshot', () => {
    const snapshot = renderer.create(<UserSearchResultItem {...props} />)
    expect(snapshot).toMatchSnapshot()
})

it('test on press called', () => {
    const { getByText } = render(<UserSearchResultItem {...props} />)

    const item = getByText(props.name)
    fireEvent.press(item)

    expect(props.onPress).toHaveBeenCalledTimes(1)
})

it('test error displays', () => {
    props.error = 'test error'
    const { getByText } = render(<UserSearchResultItem {...props} />)

    const item = getByText(props.error)
    expect(item).toBeTruthy()
})

it('test matches snapshot while loading displays', () => {
    props.loading = true
    const container = renderer.create(<UserSearchResultItem {...props} />)

    expect(container).toMatchSnapshot()
})
