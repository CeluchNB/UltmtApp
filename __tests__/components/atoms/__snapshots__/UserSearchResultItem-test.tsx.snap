// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#CEC6B4",
            "fontSize": 41.0958904109589,
            "fontWeight": "900",
          }
        }
      >
        first last
      </Text>
      <Text
        style={
          {
            "color": "#ffd600",
            "fontSize": 30.821917808219176,
            "fontWeight": "900",
          }
        }
      >
        firstlast
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "height": 40,
          "margin": 6,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 40,
        }
      }
      testID="icon-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "justifyContent": "center",
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 20,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "#40dd50",
                  "fontSize": 24,
                },
                [
                  {
                    "lineHeight": 24,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            Û∞êï
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`test matches snapshot while loading displays 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#CEC6B4",
            "fontSize": 41.0958904109589,
            "fontWeight": "900",
          }
        }
      >
        first last
      </Text>
      <Text
        style={
          {
            "color": "#ffd600",
            "fontSize": 30.821917808219176,
            "fontWeight": "900",
          }
        }
      >
        firstlast
      </Text>
    </View>
    <ActivityIndicator
      animating={true}
      color="#CEC6B4"
      size="small"
      testID="loading-spinner"
    />
  </View>
</View>
`;
